File: day8/day8api/src/main/java/bbluv/code/day8api/entitties/Employee.java
package bbluv.code.day8api.entitties;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "employees")
@Data
public class Employee {
    @Id
    private String code;
    private String name;
    private String password;
    private int gender;
    @ManyToOne
    @JoinColumn(name = "department_id", nullable = false)
    Department department;
}

File: day8/day8mobile/lib/models/employee.dart
import 'package:day8mobile/models/department.dart';

class Employee {
  String? code;
  String? name;
  String? password;
  num? gender;
  Department? department;

  Employee({this.code, this.name, this.password, this.gender, this.department});

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    map["code"] = code;
    map["name"] = name;
    map["password"] = password;
    map["gender"] = gender;
    if (department != null) {
      map["department"] = department?.toJson();
    }
    return map;
  }

  Employee.fromJson(dynamic json){
    code = json["code"];
    name = json["name"];
    password = json["password"];
    gender = json["gender"];
    department =
    json["department"] != null ? Department.fromJson(json["department"]) : null;
  }
}